arcgis-ios-sdk-samples/Utility Network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -0,0 +1,261 @@
+        utilityNetwork.trace(with: parameters) { [weak self] (traceResults, error) in
+            guard let self = self else { return }
+            if let error = error {
+                self.setStatus(message: "Trace failed.")
+                self.presentAlert(error: error)
+                SVProgressHUD.dismiss()
+                return
+            }
+            guard let elementTraceResult = traceResults?.first as? AGSUtilityElementTraceResult, !elementTraceResult.elements.isEmpty else {
+                self.setStatus(message: "Trace completed with no output.")
+                SVProgressHUD.dismiss()
+                return
+            }
+            self.clearLayersSelection()
+            SVProgressHUD.show(withStatus: "Trace completed. Selecting features…")
+            let groupedElements = Dictionary(grouping: elementTraceResult.elements) { $0.networkSource.name }
+            
+            let selectionGroup = DispatchGroup()
+            
+            groupedElements.forEach { (networkName, elements) in
+                guard let layer = self.mapView.map?.operationalLayers.first(where: { ($0 as? AGSFeatureLayer)?.featureTable?.tableName == networkName }) as? AGSFeatureLayer else { return }
+                
+                selectionGroup.enter()
+                self.utilityNetwork.features(for: elements) { [weak self, layer] (features, error) in
+                    defer {
+                        selectionGroup.leave()
+                    }
+                    if let error = error {
+                        self?.setStatus(message: "Selecting features failed.")
+                        self?.presentAlert(error: error)
+                        return
+                    }
+                    guard let features = features else { return }
+                    layer.select(features)
+                }
+            }
+            
+            selectionGroup.notify(queue: .main) { [weak self] in
+                self?.setStatus(message: "Trace completed.")
+                SVProgressHUD.dismiss()
+            }
+        }
+    }
+    
+    @IBAction func categoryButtonTapped(_ button: UIBarButtonItem) {
+        let alertController = UIAlertController(title: "Choose a category for filter barrier.", message: nil, preferredStyle: .actionSheet)
+        filterBarrierCategories.forEach { category in
+            let action = UIAlertAction(title: category.name, style: .default) { [unowned self] _ in
+                self.selectedCategory = category
+                self.setStatus(message: "\(category.name) selected.")
+            }
+            alertController.addAction(action)
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
+        alertController.popoverPresentationController?.barButtonItem = categoryButton
+        present(alertController, animated: true, completion: nil)
```

Comment:
```suggestion
        present(alertController, animated: true)
```
Remove redundant completion argument.